### DISCUSSION TITLE
No bin files for ESP32-S2-WROVER?

### DISCUSSION THREAD

**Discussion by covalspace** [micropython/micropython#12356] [ESP32]
I tried the micropython firmware for ESP32-S2 on my ESP32-S2-WROVER and noticed the pins for the LED did not align properly. 

In arduino they work fine: 

#define B_PIN 4
#define G_PIN 5
#define R_PIN 6

But with micropython they're all wonky: Blue is green and I can't find red anywhere. 

B_PIN = 4  
G_PIN = 5  
R_PIN = 6

**Comment by robert-hh** [comment]
Which board do you use? Please tell us s link to the product?

**Comment by covalspace** [comment]
I'm using flipper's wifi dev board:

https://docs.flipper.net/development/hardware/wifi-developer-board/schematics#h42h9

I'm realizing better now what's going on with the LED (this is with ESP32-S2 bin file). For some reason off/on doesn't work and on is 0 off is 1:

B_PIN = 4
G_PIN = 5
R_PIN = 6

blue_led = machine.Pin(B_PIN,machine.Pin.OUT)
green_led = machine.Pin(G_PIN,machine.Pin.OUT)
red_led = machine.Pin(R_PIN,machine.Pin.OUT)

#off
blue_led.value(1)
green_led.value(1)
red_led.value(1)
time.sleep(0.5)
#red
blue_led.value(1)
green_led.value(1)
red_led.value(0)
time.sleep(0.5)
#red+blue
blue_led.value(0)
green_led.value(1)
red_led.value(0)
time.sleep(0.5)
#blue
blue_led.value(0)
green_led.value(1)
red_led.value(1)
time.sleep(0.5)
#green+blue
blue_led.value(0)
green_led.value(0)
red_led.value(1)
time.sleep(0.5)
#green
blue_led.value(1)
green_led.value(0)
red_led.value(1)
time.sleep(0.5)

**Reply by jimmo** [reply]
This is the firmware you should be using https://micropython.org/download/ESP32_GENERIC_S2/
(There used to be a s2 wrover firmware but it was identical to the one i linked to so they were combined)

This is quite common for LEDs to be wired as "active-low", where the pin sinks current from the LED rather than sourcing it.

You can use machine.Signal solve this:

```
blue_led = machine.Signal(machine.Pin(B_PIN,machine.Pin.OUT), invert=True)
```

**Comment by covalspace** [comment]
thanks everyone
